File: CivitasRuby/casilla.rb
# encoding:utf-8
require_relative 'mazo_sorpresas'
require_relative 'jugador'
require_relative 'titulo_propiedad'

module Civitas
class Casilla
  
  # Constructor
  def initialize(nombre)
    @nombre = nombre
  end
  
  private :initialize
  
  # Métodos
  def informe(actual, todos)
    if jugador_correcto(actual, todos)
      nombre_jugador = todos[actual].nombre
      evento = "El jugador " + nombre_jugador + " ha caido en la casilla " + to_string
      Diario.instance.ocurre_evento(evento)
    end
  end
    
  def jugador_correcto(actual, todos)
    if actual < todos.length && actual >= 0
      return true
    else
      return false
    end
  end
  
  def recibe_jugador(actual, todos)
    informe(actual, todos)
  end
          
  def to_string
    return @nombre
  end
  
  # Consultor
  attr_reader :nombre, :tituloPropiedad
end
end

File: CivitasJava/civitas/Casilla.java
package civitas;

import java.util.ArrayList;

public class Casilla {
    
    // Atributos
    private String nombre;
    
    /**
     * Constructor
     * @param s nombre de la casilla 
     */
    Casilla ( String s ){
        nombre = s;
    }
          
    protected void informe (int iactual, ArrayList<Jugador> todos){
        if (jugadorCorrecto(iactual, todos)){
            String nombreJugador = todos.get(iactual).getNombre();
            String evento = "El jugador " + nombreJugador + " ha caído en la casilla " + nombre;
            Diario.getInstance().ocurreEvento(evento);
        }
    }
    
    public boolean jugadorCorrecto(int iactual, ArrayList<Jugador> todos){
        if (iactual < todos.size() && iactual >= 0)
            return true;
        else
            return false;
    }
    
    void recibeJugador (int iactual, ArrayList<Jugador> todos) throws Exception{
        informe(iactual, todos);
    }
        
    @Override
    public String toString(){
        return nombre;
    }
}
